### PR TITLE
assert_error for tests

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: omit

### CODE REVIEW

File: conans/test/utils/tools.py [hunk 1]
```diff
@@ -621,7 +621,7 @@ def run(self, command_line, user_io=None, ignore_error=False, should_error=False
             )
             raise Exception(exc_message)
 
-        if should_error and not error:
+        if assert_error and not error:
```
Comment (jgsogo) [inline conans/test/utils/tools.py]: This two clauses may be combined into a single one: `if assert_error != error`, and keep the same output for both (changing at least the header `Command failed|not failed`)

File: conans/test/remove_subsetting_test.py [hunk 2]
```diff
@@ -76,7 +74,8 @@ def build(self):
         build_folder = os.path.join(client.current_folder, "build")
         mkdir(build_folder)
         client.current_folder = build_folder
-        client.run('install .. -s os=Windows -s compiler="Visual Studio" -s compiler.version=15 -s arch=x86')
+        client.run('install .. -s os=Windows -s compiler="Visual Studio" -s compiler.version=15 '
+                   '-s arch=x86')
         # This raised an error because build_type wasn't defined
```
Comment (jgsogo) [inline conans/test/remove_subsetting_test.py]: Check the comment: `this raised an error`, but noone in the test is handling that error (neither before, nor now)...

File: conans/test/utils/tools.py (hunk 1 above)
Comment (memsharded) [reply conans/test/utils/tools.py]: error might be an integer with different values

Comment (jgsogo) [reply conans/test/utils/tools.py]: 👍  ok, then `if (assert_error and not error) or (not assert_error and error)` (same as now), but what about merging the two blocks into one?

**Pull request merged**